packages/holocron/src/holocronModule.jsx

```diff
@@ -97,57 +97,46 @@ export default function holocronModule({
   options = {},
 } = {}) {
   return function wrapWithHolocron(WrappedComponent) {
-    class HolocronModuleWrapper extends React.Component {
-      constructor(props) {
-        super(props);
-        this.state = {
-          loadCount: 0,
-          status: 'loading',
-        };
-        this.mounted = false;
-      }
-
-      componentDidMount() {
-        this.mounted = true;
-        this.initiateLoad(0, this.props);
-      }
-
-      // ignoring to support deprecated componentWillReceiveProps.
-      // This needs to be removed to support React17
-      // eslint-disable-next-line camelcase
-      UNSAFE_componentWillReceiveProps(nextProps) {
-        const { loadCount } = this.state;
-        const { props } = this;
-        if (shouldModuleReload && shouldModuleReload(props, nextProps)) {
-          const newLoadCount = loadCount + 1;
-          this.setState({ status: 'loading', loadCount: newLoadCount });
-          this.initiateLoad(newLoadCount, nextProps);
-        }
-      }
-
-      // Ignoring as internal state cannot be checked when unmounted.
-      /* istanbul ignore next */
-      componentWillUnmount() {
-        this.mounted = false;
-      }
+    const HolocronModuleWrapper = (props) => {
+      const [{ loadCount, status }, setState] = useState({ loadCount: 0, status: 'loading' });
+      const isMounted = useRef(false);
+      const prevPropsRef = useRef({});
 
-      initiateLoad(loadCount, frozenProps) {
-        return executeLoadingFunctions({
+      const initiateLoad = (currentLoadCount, frozenProps) => {
+        executeLoadingFunctions({
           loadModuleData,
           WrappedComponent,
           frozenProps,
-          loadCount,
+          currentLoadCount,
           hocName: getModuleName(WrappedComponent, name),
-          hocInstance: this,
+          hocInstance: { mounted: isMounted.current, loadCount, setState },
         });
+      };
+
+      if (
+        Object.keys(prevPropsRef.current).length > 0
+        && typeof shouldModuleReload === 'function'
+        && shouldModuleReload(prevPropsRef.current, props)
+      ) {
+        const newLoadCount = loadCount + 1;
+        setState((prevState) => ({ ...prevState, loadCount: newLoadCount }));
+        initiateLoad(newLoadCount, props);
       }
 
-      render() {
-        const { status } = this.state;
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        return <WrappedComponent {...this.props} moduleLoadStatus={status} />;
-      }
-    }
+      prevPropsRef.current = props;
+
+      React.useEffect(() => {
+        isMounted.current = true;
+        initiateLoad(loadCount, props);
```

Comment:
```suggestion
        initiateLoad(0, props);
```
I know that `loadCount` is 0 when its initialized, but hard-coding 0 here keeps it in line with the old functionality, just in case

Reply:
We are calling `initiateLoad` on mount, and `loadCount` is initialized to be 0(in react way, it should not matter). But, I have seen it to be hardcoded on old implementation though.